src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -377,6 +385,9 @@ RouterModifyTaskForShardInterval(Query *originalQuery, ShardInterval *shardInter
 	{
 		RelationRestriction *restriction = lfirst(restrictionCell);
 		List *originalBaserestrictInfo = restriction->relOptInfo->baserestrictinfo;
+		InstantiateQualWalker *instantiateQualWalker =
+			palloc0(sizeof(InstantiateQualWalker));
```

Comment:
> Why allocate separately instead of the stack?

Mostly because the existing examples in the code base uses such pattern. But, I don't have any preference so switching to stack.